src/replica/duplication/replica_follower.cpp

```diff
@@ -104,27 +105,77 @@ void replica_follower::async_duplicate_checkpoint_from_master_replica()
 
                       FAIL_POINT_INJECT_F("duplicate_checkpoint_failed",
                                           [&](string_view s) -> void { return; });
-
-                      update_master_replica_config_callback(err, std::move(resp));
+                      if (update_master_replica_config(err, std::move(resp)) == ERR_OK) {
+                          copy_master_replica_checkpoint();
+                      }
                   });
               });
 }
 
-// todo(jiashuo1)
-void replica_follower::update_master_replica_config_callback(
-    error_code err, configuration_query_by_index_response &&resp)
+error_code
```

Comment:
I have a suggestion that you can add the thread pool those functions running on.

Reply:
done